api/services/InvitationService.js

```diff
@@ -137,12 +137,57 @@ module.exports = {
     })
   },
 
-  use: (userId, invitationToken) => {
-    return Invitation.where({token: invitationToken}).fetch()
-    .then(invitation => {
-      if (!invitation) throw new Error('not found')
-      if (invitation.isExpired()) throw new Error('expired')
-      return invitation.use(userId)
-    })
+  check: (userId, token, betaAccessCode) => {
+    if (betaAccessCode) {
+      return Community.query(qb => {
```

Comment:
please de-dupe the code that's here and also on line 164

Reply:
will do